include/download_client.rst

```diff
@@ -3,50 +3,53 @@
-The image can be of any form, for example, a delta image or a full image with many merged HEX files.
-There is no revision management or specification of the flash address where the image is downloaded.
+The file is downloaded in fragments of configurable size
+(:option:`CONFIG_NRF_DOWNLOAD_MAX_FRAGMENT_SIZE`), that are returned to the
+application via events (:cpp:member:`DOWNLOAD_CLIENT_EVT_FRAGMENT`).
 
-The size of the object to download is obtained from the server.
-The object is then downloaded in fragments of a maximum fragment size (:option:`CONFIG_NRF_DOWNLOAD_MAX_FRAGMENT_SIZE`).
-For example, if the size of the object to be downloaded is 60478 bytes and the maximum fragment size is 1024 bytes, the module downloads 60 fragments of the object: 59 fragments with a size of 1024 bytes and one fragment with a size of 62 bytes.
-If any of the download requests fail, they can be repeated.
+The library can detect the size of the file being downloaded and sends an
+event (:cpp:member:`DOWNLOAD_CLIENT_EVT_DONE`) to the application when the
+download has completed.
 
-After every fragment download, a :c:type:`download_client_evt` event is returned, indicating whether the download was successful (:cpp:member:`DOWNLOAD_CLIENT_EVT_FRAGMENT`), failed (:cpp:member:`DOWNLOAD_CLIENT_EVT_ERROR`), or is completed (:cpp:member:`DOWNLOAD_CLIENT_EVT_DONE`).
+The library can detect when the server has closed the connection and returns
+an event to the application with an appropriate error code when that happens.
+The application can then resume the download by calling the
+:cpp:func:`download_client_connect` and :cpp:func:`download_client_start`
+functions again.
 
-Make sure to configure :option:`CONFIG_NRF_DOWNLOAD_MAX_RESPONSE_SIZE` in a way that suits your application.
-A low value causes many download requests, which might result in too much protocol overhead, while a large value requires a lot of RAM.
+The download happens in a separate thread, which can be paused and resumed.
+
+Make sure to configure :option:`CONFIG_NRF_DOWNLOAD_MAX_FRAGMENT_SIZE` in a way
+that suits your application. A small fragment size results in more download
+requests, and thus a higher protocol overhead, while large fragments require
+more RAM.
 
 
 Protocols
 *********
 
-The download protocol is determined from the download URI.
-Currently, only HTTP is supported.
+The library supports HTTP and HTTPS over IPv4 and IPv6.
+
 
 HTTP
 ====
 
 For HTTP, the following requirements must be met:
 
-* The address family is IPv4.
-* TCP transport is used to communicate with the server.
 * The application protocol to communicate with the server is HTTP 1.1.
 * IETF RFC 7233 is supported by the HTTP Server.
-* :option:`CONFIG_NRF_DOWNLOAD_MAX_RESPONSE_SIZE` is configured so that it can contain the entire HTTP response.
+* :option:`CONFIG_NRF_DOWNLOAD_MAX_RESPONSE_SIZE` is configured so that it
+  can contain the entire HTTP response.
 
-HTTPS is not supported.
+HTTPS
```

Comment:
Maybe add a note that only TLS 1.2 is supported?

Reply:
@lemrey ? @rlubos FYI

Reply:
Could be a good point, but I'd rather avoid another CI round, unless absolutely necessary. Docs will be a subject of change after we have RC1 tagged anyway.

Reply:
Added, thanks.

Reply:
Well, we do run CI now, so we could fix this too if needed

Reply:
Ah, ok, I saw it was added.